packages/eslint-plugin-formatjs/rules/enforce-id.ts

```diff
@@ -92,9 +92,10 @@ Actual: ${id}`,
                   ` id="${correctId}"`
                 )
               }
+              const quote = defaultMessage.indexOf('\'') === -1 ? "'" : '"';
               return fixer.replaceText(
                 messagePropNode as any,
-                `defaultMessage: '${defaultMessage}', id: '${correctId}'`
+                `defaultMessage: ${quote}${defaultMessage}${quote}, id: '${correctId}'`
```

Comment:
And escaping just single quotes in the string. Now we are good with mixed quotes.

```suggestion
                `defaultMessage: '${defaultMessage.split(`'`).join(`\\'`)}', id: '${correctId}'`
```

Reply:
thanks! this would work